src/building/bootstrapping.md

```diff
@@ -78,6 +78,47 @@ contribution [here][bootstrap-build].
 
 ## Stages of bootstrap
 
+Like most other bootstrapping compilers, `rustc` is compiled in stages.
+_Unlike_ most other compilers, where `stage0` refers to the bootstrap compiler,
+`stage0` refers to the first compiler built by bootstrap. So the following command:
+
+```sh
+x.py build --stage 0 src/rustc
+```
```

Comment:
fwiw, I feel like this doesn't seem correct. I call the stage0 compiler the beta compiler which appears in the `stage0` directory which is used during stage0.  It creates artifacts in the stage0-* directories, which during stage1 uses `Assemble` to create the stage1 compiler.  There's a phrase in the bootstrapping guide (which I may have written) which I think sums it up succinctly:

> Stage 0 uses the stage0 compiler to create stage0 artifacts which will later be uplifted to stage1.

It seems to me a lot of confusion stems from the `--stage` command-line flag, and that the terminology for "the stage" and "the compiler used in the stage" and "the compiler created by the stage" are all sort of the same.  Another problem I think is that a stage's output is written to directories with the same stage, but not actually used from there (they get copied out into stageN+1), so it straddles the boundary.  Also, people have different preconceived notions, and the numbering often seems off-by-one (https://github.com/rust-lang/rust/issues/57963 and https://github.com/rust-lang/rust/issues/59864).   But just changing that in the documentation I think will make things worse.